lib/tests/rule_actor.h

```diff
@@ -0,0 +1,151 @@
+                        std::string ruleName;
+                        try {
+                            cxxtools::SerializationInfo alertSi;
+                            JSON::readFromString(ruleJson, alertSi);
+                            alertSi.getMember(0).getMember("rule_name").getValue(ruleName);
+                        } catch (const std::exception& e) {
+                            std::cout << "Error parsing json: " << e.what() << std::endl;
+                        }
+                        if (ruleName.empty()) {
+                            zmsg_addstr(reply, "ERROR");
+                            zmsg_addstr(reply, "rule name not defined");
+                            std::cout << "== rule actor request add rule ERROR: rule name not defined" << std::endl;
+                        }
+                        if (rulesMap->find(ruleName) == rulesMap->end()) {
+                            (*rulesMap)[ruleName] = ruleJson;
+                            zmsg_addstr(reply, "OK");
+                            std::cout << "== rule actor request add rule OK: " << ruleName << std::endl;
+                        }
+                        else {
+                            zmsg_addstr(reply, "ERROR");
+                            zmsg_addstr(reply, "rule already exists");
+                            std::cout << "== rule actor request add rule ERROR: rule already exists" << std::endl;
+                        }
+                    }
+                    else {
+                        // Update rule
+                        std::string ruleName = zmsg_popstdstr(msg);
+                        std::cout << "== rule actor request update rule: " << ruleName <<  std::endl;
+                        if (rulesMap->find(ruleName) != rulesMap->end()) {
+                            (*rulesMap)[ruleName] = ruleJson;
+                            zmsg_addstr(reply, "OK");
+                            std::cout << "== actor request update rule OK" << std::endl;
+                        }
+                        else {
+                            zmsg_addstr(reply, "ERROR");
+                            zmsg_addstr(reply, "rule not found");
+                            std::cout << "== rule actor request update rule ERROR: rule not found" << std::endl;
+                        }
+                    }
+                    std::cout << "== rule actor reply" << std::endl;
+                    int r = mlm_client_sendto(client, sender, subject, NULL, 1000, &reply);
+                    zmsg_destroy(&reply);
+                    REQUIRE(r == 0);
+                }
+                else if (reqCmd == "GET") {
+                    std::string ruleName = zmsg_popstdstr(msg);
+                    std::cout << "== rule actor request '" << reqCmd << "' rule name: " << ruleName << std::endl;
+                    zmsg_t* reply = zmsg_new();
+                    REQUIRE(reply);
+                    //std::map<std::string, std::string> rulesMap;
```

Comment:
remove comment
use an iterator var. it = (rulesMap->find(ruleName)
VS (*rulesMap)[ruleName]

Reply:
Taken into account